### PR TITLE
Enable nullability in DataGridViewDataErrorEventArgs

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Enable nullability in DataGridViewDataErrorEventArgs.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/7438)

### CODE REVIEW

**Review by dreddy-work — DISMISSED** [review]
LGTM.

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridView.Methods.cs [hunk 1]
```diff
@@ -4021,10 +4022,11 @@ public bool CommitEdit(DataGridViewDataErrorContexts context)
                                 return null;
                             }
 
-                            DataGridViewDataErrorEventArgs dgvdee = new DataGridViewDataErrorEventArgs(null,
-                                                                                                       _ptCurrentCell.X,
-                                                                                                       _ptCurrentCell.Y,
-                                                                                                       context)
+                            DataGridViewDataErrorEventArgs dgvdee = new(
+                                null,
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/DataGridView.Methods.cs:4026]: [nit]
```suggestion
                                exception: null,
```

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridView.Methods.cs [hunk 2]
```diff
@@ -3966,12 +3968,11 @@ public bool CommitEdit(DataGridViewDataErrorContexts context)
...
                         }
 
-                        DataGridViewDataErrorEventArgs dgvdee = new DataGridViewDataErrorEventArgs(null,
-                                                                                    _ptCurrentCell.X,
-                                                                                    _ptCurrentCell.Y,
-                                                                                    // null,
-                                                                                    // null,
-                                                                                    context)
+                        DataGridViewDataErrorEventArgs dgvdee = new(
+                            null,
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/DataGridView.Methods.cs:3972]: [nit]
```suggestion
                            exception: null,
```

**Review by RussKie — APPROVED** [review]

**Review by RussKie — APPROVED** [review]

**Comment by RussKie** [comment]
Thank you

**Pull request merged**